rest/route/host_spawn.go

```diff
@@ -344,6 +367,142 @@ func (h *hostStartHandler) Run(ctx context.Context) gimlet.Responder {
 	return gimlet.NewJSONResponse(struct{}{})
 }
 
+////////////////////////////////////////////////////////////////////////
+//
+// POST /rest/v2/volumes
+
+const (
+	volumeDefaultProvider = evergreen.ProviderNameEc2OnDemand
+	volumeDefaultRegion   = evergreen.DefaultEC2Region
+)
+
+type createVolumeHandler struct {
+	sc data.Connector
+
+	Type string `json:"type"`
+	Size int    `json:"size"`
+}
+
+func makeCreateVolume(sc data.Connector) gimlet.RouteHandler {
+	return &createVolumeHandler{
+		sc: sc,
+	}
+}
+
+func (h *createVolumeHandler) Factory() gimlet.RouteHandler {
+	return &createVolumeHandler{
+		sc: h.sc,
+	}
+}
+
+func (h *createVolumeHandler) Parse(ctx context.Context, r *http.Request) error {
+	return errors.WithStack(util.ReadJSONInto(r.Body, h))
+}
+
+func (h *createVolumeHandler) Run(ctx context.Context) gimlet.Responder {
+	u := MustHaveUser(ctx)
+
+	volume := &host.Volume{
+		CreatedBy: u.Id,
+		Type:      h.Type,
+		Size:      h.Size,
+	}
+
+	// TODO: Allow different providers/regions
+	mgrOpts := cloud.ManagerOpts{
+		Provider: volumeDefaultProvider,
+		Region:   volumeDefaultRegion,
+	}
+	mgr, err := cloud.GetManager(ctx, mgrOpts, evergreen.GetEnvironment().Settings())
```

Comment:
I think there's an env that you can pass into the route construtor to avoid doing the getenv/settings dance.